jumanji/environments/combinatorial/cvrp/env_test.py

```diff
@@ -0,0 +1,305 @@
+from jumanji.environments.combinatorial.cvrp.utils import (
+    DEPOT_IDX,
+    compute_tour_length,
+    get_augmentations,
+)
+
+from jumanji.testing.env_not_smoke import check_env_does_not_smoke
+from jumanji.testing.pytrees import assert_is_jax_array_tree
+from jumanji.types import StepType, TimeStep
+
+
+@pytest.fixture
+def cvrp_env() -> CVRP:
+    """Instantiates a default CVRP environment."""
+    return CVRP()
+
+
+def test_cvrp__reset(cvrp_env: CVRP) -> None:
+    """Validates the jitted reset of the environment."""
+    reset_fn = jax.jit(cvrp_env.reset)
+    key = random.PRNGKey(0)
+    state, timestep = reset_fn(key)
+
+    assert isinstance(timestep, TimeStep)
+    assert isinstance(state, State)
+
+    # Initial position is at depot, so current capacity is max capacity
+    assert (state.capacity == cvrp_env.max_capacity)
+    # # The depot is initially visited
+    assert state.visited_mask[DEPOT_IDX] == 1
+    assert state.visited_mask.sum() == 1
+    # First visited position is the depot
+    assert state.order[0] == DEPOT_IDX
+    assert state.num_total_visits == 1
+    # Cost of the depot must be 0
+    assert state.problem[DEPOT_IDX, 2] == 0
+
+    assert_is_jax_array_tree(state)
+
+
+def test_cvrp__step(cvrp_env: CVRP) -> None:
+    """Validates the jitted step of the environment."""
+    chex.clear_trace_counter()
+
+    step_fn = chex.assert_max_traces(cvrp_env.step, n=1)
+    step_fn = jax.jit(step_fn)
+
+    key = random.PRNGKey(0)
+    state, timestep = cvrp_env.reset(key)
+
+    # Starting position is depot, new action to visit first node
+    new_action = 1
+    new_state, next_timestep = step_fn(state, new_action)
+
+    # Check that the state has changed
+    assert not jnp.array_equal(new_state.position, state.position)
+    assert not jnp.array_equal(new_state.capacity, state.capacity)
+    assert not jnp.array_equal(new_state.visited_mask, state.visited_mask)
+    assert not jnp.array_equal(new_state.order, state.order)
+    assert not jnp.array_equal(new_state.num_total_visits, state.num_total_visits)
+
+    # Check that the state is made of DeviceArrays, this is false for the non-jitted
+    # step function since unpacking random.split returns numpy arrays and not device arrays.
+    assert_is_jax_array_tree(new_state)
+
+    # Check the state was changed as expected
+    assert new_state.visited_mask[new_action] == 1
+    assert new_state.visited_mask.sum() == 1
+
+    # New step with same action should be invalid
+    state = new_state
+
+    new_state, next_timestep = step_fn(state, new_action)
+
+    # Check that the state has not changed
+    assert jnp.array_equal(new_state.position, state.position)
+    assert jnp.array_equal(new_state.visited_mask, state.visited_mask)
+    assert jnp.array_equal(new_state.order, state.order)
+    assert jnp.array_equal(new_state.num_total_visits, state.num_total_visits)
+
+
+def test_cvrp__does_not_smoke(cvrp_env: CVRP, capsys: pytest.CaptureFixture) -> None:
```

Comment:
Why do we have the capture fixture here?
```suggestion
def test_cvrp__does_not_smoke(cvrp_env: CVRP) -> None:
```